js/src/TimeSeriesBar.js

```diff
@@ -0,0 +1,636 @@
+/**
+ * TimeSeriesBar
+ *
+ */
+
+var utils = require('./utils')
+var CallbackManager = require('./CallbackManager')
+var _ = require('underscore')
+var data_styles = require('./data_styles.js')
+
+var TimeSeriesBar = utils.make_class()
+
+var builder
```

Comment:
global variables should be converted to attributes of the TimeSeriesBar. Set them up in `init` function by assigning them to `this`.